Submitted By:            Pierre Labastie <pierre dot labastie at neuf dot fr>
Date:                    2020-03-04
Initial Package Version: 1.22.4.0
Upstream Status:         PR submitted
Origin:                  Debian patches
Descripton:              Fix some headers moved around by Mesa-20.x.

diff --git a/configure.ac b/configure.ac
index da188edd721eb4884683c8ef7263a25cd4429851..b0b0c94d8945d095b7b31728fe5482e622599d96 100644
--- a/configure.ac
+++ b/configure.ac
@@ -670,10 +670,15 @@ AS_IF([test "x$enable_gles1" = "xyes"],
             AC_SUBST([COGL_EGL_INCLUDES])
 
             AC_CHECK_HEADERS([EGL/eglext.h],
-                             [COGL_EGL_INCLUDES="$COGL_EGL_INCLUDE
+                             [COGL_EGL_INCLUDES="$COGL_EGL_INCLUDES
 #include <EGL/eglext.h>"],
                              [],
                              [$COGL_EGL_INCLUDES])
+            AC_CHECK_HEADERS([EGL/eglmesaext.h],
+                             [COGL_EGL_INCLUDES="$COGL_EGL_INCLUDES
+#include <EGL/eglmesaext.h>"],
+                             [],
+                             [$COGL_EGL_INCLUDES])
 
             # Check for a GLES 1.x Common Profile library with/without EGL.
             #
@@ -1178,6 +1183,12 @@ AS_IF([test "x$NEED_EGL" = "xyes" && test "x$EGL_CHECKED" != "xyes"],
               [],
               [AC_MSG_ERROR([Unable to locate required EGL headers])],
               [#include <EGL/egl.h>])
+            AC_CHECK_HEADERS(
+              [EGL/eglmesaext.h],
+              [],
+              [AC_MSG_ERROR([Unable to locate required EGL headers])],
+              [#include <EGL/egl.h>
+#include <EGL/eglext.h>])
 
             AC_CHECK_LIB(EGL, [eglInitialize],
               [COGL_EXTRA_LDFLAGS="$COGL_EXTRA_LDFLAGS -lEGL"],
@@ -1188,7 +1199,8 @@ AS_IF([test "x$NEED_EGL" = "xyes" && test "x$EGL_CHECKED" != "xyes"],
           )
 
         COGL_EGL_INCLUDES="#include <EGL/egl.h>
-#include <EGL/eglext.h>"
+#include <EGL/eglext.h>
+#include <EGL/eglmesaext.h>"
         AC_SUBST([COGL_EGL_INCLUDES])
       ])
 
